idfmt.go

```diff
@@ -0,0 +1,42 @@
+package iface
+
+import (
+	"fmt"
+
+	peer "github.com/libp2p/go-libp2p-core/peer"
+	mbase "github.com/multiformats/go-multibase"
+)
+
+func VerifyIDFormatLabel(formatLabel string) error {
+	switch formatLabel {
+	case "b58mh":
+		return nil
+	case "b36cid":
+		return nil
+	}
+	return fmt.Errorf("invalid output format option")
+}
+
+func FormatID(id peer.ID, formatLabel string) string {
+	switch formatLabel {
+	case "b58mh":
+		return id.Pretty()
+	case "b36cid":
+		if s, err := peer.ToCid(id).StringOfBase(mbase.Base36); err != nil {
+			panic(err)
+		} else {
+			return s
+		}
+	default:
+		panic("unreachable")
+	}
+}
```

Comment:
I'd like to have less notation rather than more, so limiting concepts lie "b58mh" or "b36cid" to the go-ipfs CLI + HTTP API for a couple commands rather than having it everywhere.

As a result I don't think this function is necessary `FormatKeyID` is all that's used in this PR, and in the go-ipfs PR https://github.com/ipfs/go-ipfs/pull/7554 it's used but only to process the CLI arguments.

Reply:
Done.